### PR TITLE
layers: Fix null descriptor set crash

### PR DESCRIPTION
**Pull request by ziga-lunarg**

### CODE REVIEW

**Review by ncesario-lunarg — APPROVED** [review]
LGTM!

**Comment by ncesario-lunarg** [comment]
Is there a test case that should be added to cover this?

File: layers/cmd_buffer_state.cpp [hunk 1]
```diff
@@ -1045,6 +1045,9 @@ void CMD_BUFFER_STATE::UpdatePipelineState(CMD_TYPE cmd_type, const VkPipelineBi
             if (set_index >= state.per_set.size()) {
                 continue;
             }
+            if (!state.per_set[set_index].bound_descriptor_set) {
+                continue;
```
Comment (arno-lunarg) [inline layers/cmd_buffer_state.cpp:1049]: Nit: this test could use the following variable

**Comment by arno-lunarg** [comment]
LGTM! (One NIT)

**Review by arno-lunarg — APPROVED** [review]

**Review by jeremyg-lunarg — APPROVED** [review]

File: layers/cmd_buffer_state.cpp (hunk 1 above)
Comment (ziga-lunarg) [reply layers/cmd_buffer_state.cpp:1049]: The following variable is a refence, if this is nullptr it is not valid to take a reference from it.

**Comment by ziga-lunarg** [comment]
> Is there a test case that should be added to cover this?

I am seeing these crashes in my own application, using multiple threads with indirect draws, but I have not been able to reproduce this with a test yet.

File: layers/cmd_buffer_state.cpp (hunk 1 above)
Comment (arno-lunarg) [reply layers/cmd_buffer_state.cpp:1049]: To my knowledge, I think it is valid to do so? `auto &descriptor_set` refers to the variable `state.per_set[set_index].bound_descriptor_set`, and his just a "synonym" for this variable, unrelated to its potential values.  Or am I missing something.
This code compiles and runs just fine: https://godbolt.org/z/xj6ejcE87

Comment (ziga-lunarg) [reply layers/cmd_buffer_state.cpp:1049]: It's not valid to have a reference to a nullptr, but this isn't even a regular pointer, it's a `shared_ptr` so I guess I was looking at the wrong code. The problem comes in the line after this.

File: layers/cmd_buffer_state.cpp [hunk 2]
```diff
@@ -1045,8 +1045,14 @@ void CMD_BUFFER_STATE::UpdatePipelineState(CMD_TYPE cmd_type, const VkPipelineBi
...
+            }
             // Pull the set node
             auto &descriptor_set = state.per_set[set_index].bound_descriptor_set;
+            if (descriptor_set) {
```
Comment (arno-lunarg) [inline layers/cmd_buffer_state.cpp:1053]: I think the `!` is missing

File: layers/cmd_buffer_state.cpp [hunk 3]
```diff
@@ -1045,8 +1045,14 @@ void CMD_BUFFER_STATE::UpdatePipelineState(CMD_TYPE cmd_type, const VkPipelineBi
             if (set_index >= state.per_set.size()) {
                 continue;
             }
+            if (!state.per_set[set_index].bound_descriptor_set) {
```
Comment (arno-lunarg) [inline layers/cmd_buffer_state.cpp:1048]: This can now be removed

**Pull request merged**